WorkspaceDiagnosticsHandlerTest.testResetPomDiagnostics fails randomly

Error Message:

No pom.xml errors were found expected:<3> but was:<6>

Stacktrace:

java.lang.AssertionError: No pom.xml errors were found expected:<3> but was:<6>
	at org.eclipse.jdt.ls.core.internal.handlers.WorkspaceDiagnosticsHandlerTest.testResetPomDiagnostics(WorkspaceDiagnosticsHandlerTest.java:404)

Error Message:
Unexpected diagnostics:
[Diagnostic [
  range = Range [
    start = Position [
      line = 0
      character = 0
    ]
    end = Position [
      line = 0
      character = 0
    ]
  ]
  severity = Warning
  code = "0"
  source = "Java"
  message = "Project configuration is not up-to-date with pom.xml, requires an update."
  relatedInformation = null
]] expected:<0> but was:<1>

Stacktrace

java.lang.AssertionError: 
Unexpected diagnostics:
[Diagnostic [
  range = Range [
    start = Position [
      line = 0
      character = 0
    ]
    end = Position [
      line = 0
      character = 0
    ]
  ]
  severity = Warning
  code = "0"
  source = "Java"
  message = "Project configuration is not up-to-date with pom.xml, requires an update."
  relatedInformation = null
]] expected:<0> but was:<1>
	at org.eclipse.jdt.ls.core.internal.handlers.WorkspaceDiagnosticsHandlerTest.testResetPomDiagnostics(WorkspaceDiagnosticsHandlerTest.java:433)

--- a/org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/WorkspaceDiagnosticsHandlerTest.java
+++ b/org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/WorkspaceDiagnosticsHandlerTest.java
@@ -394,7 +394,7 @@ public class WorkspaceDiagnosticsHandlerTest extends AbstractProjectsManagerBase
 
 		ResourceUtils.setContent(pom, ResourceUtils.getContent(pom).replaceAll("<profiles>", compilerPlugin + "\n<profiles>"));
 
-		ResourcesPlugin.getWorkspace().build(IncrementalProjectBuilder.INCREMENTAL_BUILD, monitor);
+		waitForBackgroundJobs();
 
 		ArgumentCaptor<PublishDiagnosticsParams> captor = ArgumentCaptor.forClass(PublishDiagnosticsParams.class);
 		verify(connection, atLeastOnce()).publishDiagnostics(captor.capture());

